### PR TITLE
Vagrant CoreOS guide for Kubernetes

### PR DESCRIPTION
**Pull request by caseydavenport**

### CODE REVIEW

**Comment by tomdee** [comment]
If you want me to review pls assign to me.

File: docs/kubernetes/vagrant-coreos/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Size of the cluster created by Vagrant
+num_instances=2
+
+# Change basename of the VM
+instance_name_prefix="calico"
+
+# Official CoreOS channel from which updates should be downloaded
+update_channel='alpha'
```
Comment (tomdee) [inline docs/kubernetes/vagrant-coreos/Vagrantfile]: Maybe switch to stable?

File: docs/kubernetes/vagrant-coreos/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  config.ssh.insert_key = false
+
+  config.vm.box = "coreos-%s" % update_channel
+  config.vm.box_url = "http://%s.release.core-os.net/amd64-usr/current/coreos_production_vagrant.json" % update_channel
+
+  config.vm.provider :virtualbox do |v|
+    # On VirtualBox, we don't have guest additions or a functional vboxsf
+    # in CoreOS, so tell Vagrant that so it can be smarter.
+    v.check_guest_additions = false
+    v.memory = 4096
```
Comment (tomdee) [inline docs/kubernetes/vagrant-coreos/Vagrantfile]: Maybe drop this a little? 2 GB and 2 cores?

File: docs/kubernetes/vagrant-coreos/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  # Set up each box
+  (1..num_instances).each do |i|
+    vm_name = "%s-%02d" % [instance_name_prefix, i]
+    config.vm.define vm_name do |host|
+      host.vm.hostname = vm_name
+
+      ip = "172.18.18.#{i+100}"
+      host.vm.network :private_network, ip: ip
+
+      if i == 1
```
Comment (tomdee) [inline docs/kubernetes/vagrant-coreos/Vagrantfile]: Drop this i==1 block

File: docs/kubernetes/vagrant-coreos/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                                            "gcr.io/google_containers/pause:0.8.0", 
+                                            "gcr.io/google_containers/skydns:2015-10-13-8c72f8c",
+                                            "gcr.io/google_containers/kube2sky:1.11",
+                                            "gcr.io/google_containers/etcd:2.2.1", 
+                                            "gcr.io/google_containers/exechealthz:1.0"]
+        host.vm.provision :file, :source => "../cloud-config/master-config-template.yaml", :destination => "/tmp/vagrantfile-user-data"
+        host.vm.provision :file, :source => "../manifests/skydns.yaml", :destination => "/home/core/skydns.yaml"
+        host.vm.provision :file, :source => "../manifests/guestbook.yaml", :destination => "/home/core/guestbook.yaml"
+        host.vm.provision :shell, :inline => "mv /tmp/vagrantfile-user-data /var/lib/coreos-vagrant/", :privileged => true
+	host.vm.network "forwarded_port", guest: 8080, host: 8080
```
Comment (tomdee) [inline docs/kubernetes/vagrant-coreos/Vagrantfile]: Drop these two forwarded ports

File: docs/kubernetes/vagrant-coreos/Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	  v.memory = 4096
+	  v.cpus = 4
+	end
+	host.vm.provision :docker, images: ["busybox:latest", 
+                                            "gcr.io/google_containers/pause:0.8.0", 
+                                            "gcr.io/google_containers/skydns:2015-10-13-8c72f8c",
+                                            "gcr.io/google_containers/kube2sky:1.11",
+                                            "gcr.io/google_containers/etcd:2.2.1", 
+                                            "gcr.io/google_containers/exechealthz:1.0"]
+        host.vm.provision :file, :source => "../cloud-config/master-config-template.yaml", :destination => "/tmp/vagrantfile-user-data"
```
Comment (tomdee) [inline docs/kubernetes/vagrant-coreos/Vagrantfile]: Move this down a couple of lines so it's next to the `mv` line

File: docs/kubernetes/cloud-config/node-config-template.yaml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+          "name": "calico-k8s-network",
+          "type": "calico",
+          "etcd_authority": "172.18.18.101:2379",
+          "log_level": "info",
+          "ipam": {
+              "type": "calico-ipam"
+          }
+      }
+
+  # Kubeconfig file.
```
Comment (tomdee) [inline docs/kubernetes/cloud-config/node-config-template.yaml]: What's this for? Is it actually needed?

**Comment by tomdee** [comment]
OK, few comments, otherwise LGTM.

File: docs/kubernetes/cloud-config/node-config-template.yaml (hunk 6 above)
Comment (caseydavenport) [inline docs/kubernetes/cloud-config/node-config-template.yaml]: This is used by the DNS service (and potentially other services that might be deployed) to access the API.

**Pull request merged**